### PR TITLE
[SPARK-47155][PYTHON] Fix Error Class Issue

### PR DESCRIPTION
**Pull request by sunan135**
### What changes were proposed in this pull request?
Make create_data_source.py use correct error class.

### Why are the changes needed?
This is part of the effort of SPARK-44076 that enable python user to develop data source in python and make python accessible to wider python community. Error class "PYTHON_DATA_SOURCE_CREATE_ERROR" and "PYTHON_DATA_SOURCE_METHOD_NOT_IMPLEMENTED" used in create_data_source.py doesn't exist and will throw error class not found error.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
 Existing test sufficient.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by xinrong-meng** [comment]
Good catch! Thanks for working on that. Would you add `[PYTHON]` to PR title?

**Review by HyukjinKwon — APPROVED** [review]
Looks good, cc @allisonwang-db

**Review by yaooqinn — APPROVED** [review]

**Review by allisonwang-db — APPROVED** [review]
Thanks for the fix!

**Comment by HyukjinKwon** [comment]
see https://github.com/sunan135/spark/actions/runs/8100089923

**Comment by chaoqin-li1123** [comment]
What may be the issue here? The workflow failure is weird. @HyukjinKwon 
`SHA: 31dfc375e56020b7c5d4b00ff0ed1c4aebd8ef2b
Error: Failed to retrieve check_run_head after 3 attempts
    at eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:35424:16), <anonymous>:92:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:35522:20)
Error: Unhandled error: Error: Failed to retrieve check_run_head after 3 attempts`

**Comment by HyukjinKwon** [comment]
I think the actions should be enabled at https://github.com/sunan135/spark/settings/actions by `Allow all actions and reusable workflows`

File: python/pyspark/sql/worker/create_data_source.py [hunk 1]
```diff
@@ -150,8 +150,8 @@ def main(infile: IO, outfile: IO) -> None:
                     is_ddl_string = True
             except NotImplementedError:
                 raise PySparkRuntimeError(
-                    error_class="PYTHON_DATA_SOURCE_METHOD_NOT_IMPLEMENTED",
-                    message_parameters={"type": "instance", "method": "schema"},
+                    error_class="NOT_IMPLEMENTED",
+                    message_parameters={"feature": "Reader.schema"},
```
Comment (HyukjinKwon) [inline python/pyspark/sql/worker/create_data_source.py:154]:
```suggestion
                    message_parameters={"feature": "DataSource.schema"},
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
```
======================================================================
FAIL [0.001s]: test_error_classes_sorted (pyspark.errors.tests.test_errors.ErrorsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/spark/spark/python/pyspark/errors/tests/test_errors.py", line 33, in test_error_classes_sorted
    self.assertTrue(
AssertionError: False is not true : Error class [DATA_SOURCE_RETURN_SCHEMA_MISMATCH] should place after [DATA_SOURCE_CREATE_ERROR].

Run 'cd $SPARK_HOME; bin/pyspark' and 'from pyspark.errors.exceptions import _write_self; _write_self()' to automatically sort them.
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**